### DISCUSSION TITLE
[BUG] - Blog post comments are not retrieving blog specific comments

### DISCUSSION THREAD

**Discussion by Ericcwong** [Shopify/storefront-api-feedback#173] [Help] [answered]
Hi everyone,

When I query for a blog it returns all the comments within the collection? It used to only return blog specific comments. Below is the query I am using.

```
 query Articles(
    $country: CountryCode
    $language: LanguageCode
    $pageBy: Int!
    $handle: String
  ) @inContext(country: $country, language: $language) {
    blog(handle: $handle) {
      articles(first: $pageBy, sortKey: PUBLISHED_AT, reverse: true) {
        pageInfo {
          endCursor
          hasNextPage
        }
        nodes {
          title
          image {
            url
            altText
          }
          #Under title description
          content
          comments(first: 9) { <------ this returns not what its supposed to.
            nodes {
              author {
                name
              }
            }
          }
          authorV2 {
            name
          }
          tags
          publishedAt
          handle
        }
      }
    }
  }
;

**Comment by johncraigcole** [comment, marked as answer]
Hey @Ericcwong sorry some time has elapsed with no response on this. Just wanted to confirm an issue has been created on our end for this. I don't have a timeline for a fix, but will follow up when I do.

**Reply by Ericcwong** [reply]
Nice! I appreciate it! Is there a ticket for creating blog comments as well by chance?@johncraigcole

**Reply by johncraigcole** [reply]
The Storefront API is generally limited in what type of data mutations are possible given it's an unauthenticated api. I don't believe there's any plans to extend the Storefront API to offer a `commentCreate` type mutation. Right now, the REST Admin API is the only way to manage blog comments via API. 

If you or anyone else are starting to get requests to manage blog comments more on custom storefronts and want to continue to leveraging Shopify's blog resource to do so, let us know. Not having a full blog comment solution for custom storefronts could just be feedback we haven't generally received to date.

**Reply by Ericcwong** [reply]
We do plan to transition to Remix eventually but unsure on when. We would appreciate it if we could get blogs to have the option to post new comments and the fix to load the blogs comments. We have customers that comment on blogs or articles basically daily. Not having that option feels like we aren't connecting as much with our customers.

Thank you so much for taking the time to help me out with this situation!

**Reply by johncraigcole** [reply]
Hey @Ericcwong the fix for loading comments from other articles has been fixed. Please let me know if you continue to experience the issue.

**Reply by Ericcwong** [reply]
Thank you for fixing that! @johncraigcole
